staging/src/k8s.io/apiserver/pkg/admission/plugin/cel/admission.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cel
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/client-go/tools/cache"
+)
+
+////////////////////////////////////////////////////////////////////////////////
+// Plugin Definition
+////////////////////////////////////////////////////////////////////////////////
+
+// Definition for CEL admission plugin. This is the entry point into the
+// CEL admission control system.
+//
+// Each plugin is asked to validate every object update.
+
+const (
+	// PluginName indicates the name of admission plug-in
+	PluginName = "CEL"
+)
+
+// Register registers a plugin
+func Register(plugins *admission.Plugins) {
```

Comment:
Feature gated?

Reply:
I was thinking the code in `kubeapiserver` which adds the plugin to the initialization chain would be feature-gated. It would also have to use the feature gate to start the admission controller and object converter together with the plugin. 

@cici37 do we want the plugin to internally enforce the feature gate logic?

Reply:
Featured gated in chain initialization should be sufficient. Thanks